Sources/SourceKitBazelBSP/RequestHandlers/BuildTargets/BazelQueryParser.swift

```diff
@@ -17,114 +17,196 @@
+                guard let path = srcMap[src] else {
+                    let error = BazelTargetParserError.noSrcFound(src)
+                    logger.info("\(error, privacy: .public)")
+                    return nil
+                }
+                return try URI(string: path)
             }
+
+            let data = SourceKitBuildTarget(
+                toolchain: try URI(string: "file://" + toolchainPath)
+            ).encodeToLSPAny()
+
+            let buildTarget = BuildTarget(
+                id: BuildTargetIdentifier(uri: id),
+                displayName: rule.name,
+                baseDirectory: baseDirectory,
+                tags: testOnly ? [.test, .library] : [.library],
+                capabilities: capabilities,
+                languageIds: isSwift ? [.swift] : [.objective_c],
+                dependencies: deps,
+                dataKind: .sourceKit,
+                data: data
+            )
+
+            result.append((buildTarget, srcs))
         }
-        return targets
+
+        return result
     }
 
-    static private func getTargetForLibrary(
-        _ childElement: XMLElement,
-        _ className: String,
-        _ rootUri: String,
-        _ toolchainPath: String
-    ) throws -> (BuildTarget, [URI])? {
-        let bazelTarget = childElement.attribute(forName: "name")?.stringValue ?? ""
-        guard bazelTarget.starts(with: "//") else {
-            // FIXME
-            return nil
-        }
-        let isSwift = className.contains("swift")
-        let fullPath = rootUri + "/" + bazelTarget.dropFirst(2)
-        let uriRaw = bazelTargetToURI(fullPath)
-        let basePath = uriRaw.components(separatedBy: "___")[0]
-        var targetSrcs: [URI] = []
-        let uri: URI = try URI(string: uriRaw)
-
-        for child in (childElement.children ?? []) {
-            if child.name != "list" { continue }
-            guard let childElement = child as? XMLElement else { continue }
-            let name = childElement.attribute(forName: "name")?.stringValue ?? ""
-            guard name == "srcs" else { continue }
-            for srcsEntry in (childElement.children ?? []) {
-                if srcsEntry.name != "label" { continue }
-                guard let srcsEntryElement = srcsEntry as? XMLElement else { continue }
-                let srcValue = srcsEntryElement.attribute(forName: "value")?.stringValue ?? ""
-                // FIXME
-                if !srcValue.starts(with: "//") { continue }
-                let src = srcValue.replacingOccurrences(of: ":", with: "/")
-                let srcUri = try URI(string: "file://" + rootUri + "/" + src.dropFirst(2))
-                targetSrcs.append(srcUri)
+    /// Bazel query outputs a list of targets and each target contains list of attributes.
+    /// The `srcs` attribute is a list of source_file labels instead of URI, thus we need
+    /// a hashmap to reduce the time complexity.
+    static func buildSourceFilesMap(
```

Comment:
The previous logic had some additional checks that are not present here (particularly filtering against only local content `//` since I hadn't played with third party repos at the time), I think we might still need those, but am not 100% sure. I guess as long as it works for you and @josh-arnold-1 then it should be fine.